src/controller/p2p/preheat/controller.go

```diff
@@ -197,14 +207,42 @@ func (c *controller) CountPolicy(ctx context.Context, query *q.Query) (int64, er
 	return c.pManager.Count(ctx, query)
 }
 
+// TriggerParam ...
+type TriggerParam struct {
+	PolicyID int64
+	Trigger  string
+}
+
 // CreatePolicy creates the policy.
-func (c *controller) CreatePolicy(ctx context.Context, schema *policyModels.Schema) (int64, error) {
+func (c *controller) CreatePolicy(ctx context.Context, schema *policyModels.Schema) (id int64, err error) {
 	if schema != nil {
 		now := time.Now()
 		schema.CreatedAt = now
 		schema.UpdatedTime = now
 	}
-	return c.pManager.Create(ctx, schema)
+
+	if err = orm.WithTransaction(func(ctx context.Context) error {
```

Comment:
I don't think the DB transaction can roll back the already created schedule job at the job service side when the scheme is failed to persist.

Maybe we can double-check if the policy with the ID that is returned in the `TriggerParam ` is existing in the DB, if not, then unscheduled the corn job?

Reply:
`orm.WithTransaction` woks in job service side, it is a wrapper for `Begin`, `Commit` `Rollback`.